worldedit-core/src/main/java/com/fastasyncworldedit/core/util/ProcessorTraverser.java

```diff
@@ -0,0 +1,48 @@
+package com.fastasyncworldedit.core.util;
+
+import com.fastasyncworldedit.core.extent.processor.BatchProcessorHolder;
+import com.fastasyncworldedit.core.extent.processor.MultiBatchProcessor;
+import com.fastasyncworldedit.core.queue.IBatchProcessor;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+public class ProcessorTraverser<T extends IBatchProcessor> {
+
+    private final T root;
+
+    public ProcessorTraverser(@Nonnull T root) {
+        this.root = root;
+    }
+
+    @SuppressWarnings("unchecked")
+    public @Nullable <U extends IBatchProcessor> U find(Class<U> clazz) {
+        try {
+            Queue<IBatchProcessor> processors = new ArrayDeque<>(5);
```

Comment:
Why 5?

Reply:
I did a quick count of the processors we almost always add, but tbh there'd realistically be no perf hit with any number